### PR TITLE
feat: table deletion protection

### PR DESCRIPTION
**Pull request by kolea2**

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java [hunk 1]
```diff
@@ -152,6 +158,10 @@ public List<ColumnFamily> getColumnFamilies() {
     return columnFamilies;
   }
 
+  public boolean isProtected() {
```
Comment (TracyCuiCan) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java:161]: nit: isDeletionProtected

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClientTests.java [hunk 2]
```diff
@@ -204,6 +212,54 @@ public void testCreateTable() {
     assertThat(result).isEqualTo(Table.fromProto(expectedResponse));
   }
 
+  @Test
+  public void testCreateTableWithDeletionProtection() {
```
Comment (TracyCuiCan) [inline google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClientTests.java:216]: This test doesn't seem to create a table with deletion protection, or maybe I'm missing something?

Comment (kolea2) [reply google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClientTests.java:216]: yes great call, thanks :D

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java (hunk 1 above)
Comment (kolea2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java:161]: This was to be consistent with go - https://github.com/googleapis/google-cloud-go/blob/main/bigtable/bttest/inmem.go#L1330. LMK your thoughts

**Pull request closed without merging**